tests/system/client.go

```diff
@@ -72,25 +77,41 @@ func waitForSession(akPath, akAddr, step string) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("invalid duration %q: %w", match[1], err)
 	}
-	id := match[2]
+	isSessionExpected := true // wait .. for .. session
+	if match[2] == "unless" {
+		isSessionExpected = false // wait .. unless .. session
+	}
+	waitType := match[2]
+	id := match[3]
```

Comment:
No need for if-else for `isSessionExpected`, and no need to split it from `stateRegex` below.

```suggestion
	id := match[3]
	waitType := match[2]
	isSessionExpected := waitType == "for"
	stateRegex = sessionStateAll
	if isSessionExpected {
		stateRegex = sessionStateFinal
	}
```